diksha/diksha.controller.js

```diff
@@ -423,51 +402,73 @@ let generateResponseStructure = (rSt, rsps) => {
     //console.log(JSON.stringify(foo, null, 4));
 
     defer.resolve({
-        responseStructure: rSt
+	    responseStructure: rSt
     });
     return defer.promise;
 }
 
-let doPrebuiltSearch = (requestSkeletons, query) => {
-    let defer = q.defer();
-    let bulkedResponses = {};
-    let bulkedResponsePromises = [];
-    let sectionNames = [];
-    let keys = Object.keys(requestSkeletons);
-    keys.forEach(key => {
-        if (typeof requestSkeletons[key] === 'undefined' || requestSkeletons[key] === null) {} else {
-            reqSkel = requestSkeletons[key];
-            reqSkel.query = query;
-            sectionNames.push(key);
-            bulkedResponsePromises.push(doThoroughSearch(reqSkel));
-        }
-    });
-    q.all(bulkedResponsePromises).then(values => {
-        let responses = {};
-
-        for (let i = 0; i < sectionNames.length; i++) {
-            responses[sectionNames[i]] = values[i].responses.map(response => response.fields);
-        }
-
-        //Object.keys(responses).forEach((key, index) => {
-        //let naam = '/home/admin/' + index;
-        //console.log('writing: ', naam);
-        //fs.writeFileSync(naam, JSON.stringify(responses[key], null, 4));
-        //console.log(index + '>' + key + ': ' + JSON.stringify(responses[key], null, 4))
-
-        //	console.log(index+1, ":", key);
-        //});
-
-        return defer.resolve({
-            responses
-        });
-    }).catch(e => {
-        console.log(e);
-        return defer.reject(e);
-    });
-    return defer.promise;
+let doSectionwiseSearch = (sectionObject) => {
+	    let searchpromise;
+	    let queryObject = {
+	      
+		   "conjuncts" : [] 
+	    }    
+	    for(let key in sectionObject){
+                
+		    if(key !== "compatibilityLevel"){
+    
+			    if(typeof sectionObject[key] === "object") {
+				    let disjuncts = [];
+				    for(let i in sectionObject[key])
+				    {	
+				        disjuncts.push({ 
+		   			        "field" : "archive.items." + key ,
+		   			        "match_phrase" : "" + sectionObject[key][i]
+				        });
+				    }	
+				    queryObject.conjuncts.push({disjuncts});
+
+			    }
+			    else {
+		   	            field = { 
+		   		            "field" : "archive.items."+ key ,
+		   		            "match_phrase" : "" + sectionObject[key]
+				        }
+			    }		
+            }   	
+	    }
+	    
+	     searchPromise = advancedSearch({
+	        indexName : pluginProfile.db,
+	        query : queryObject
+	    });
+		return searchPromise;
+}	
+
+let getAllPromises = (responsePromise) => {	
+        let defer = q.defer(); 
+	    let hitPromises = [];
+        let hits = JSON.parse(responsePromise.body).hits;
+	    let total_hits = JSON.parse(responsePromise.body).total_hits;	    
+            //console.log('hits found : '+ total_hits);
```

Comment:
Remove these logs

Reply:
And other unnecessary ones, along with unused code

Reply:
Done